controllers/user.controller.js

```diff
@@ -0,0 +1,106 @@
+const userService = require("../services/user.service");
+
+const {
+  isValidEmailSyntax,
+  isValidPassword,
+  composite,
+  compare,
+  getRandString,
+  getPublicUserData,
+  respondWithError,
+  respondWithStatus,
+} = require("../services/utils");
+
+const bcrypt = require("bcrypt");
+
+const { SERCURITY, ERRORS, CRUD_OPS } = require("../constants");
+
+const register = async (req, res) => {
+  try {
+    const { name, email, password } = req.body;
+    if (!isValidEmailSyntax(email)) {
+      return respondWithError(res, ERRORS.INVALID_EMAIL_ADDRESS);
+    }
+    if (!isValidPassword(password)) {
+      return respondWithError(res, ERRORS.INVALID_PASSWORD);
+    }
+    if (await userService.isExistingUser(name)) {
+      return respondWithError(res, ERRORS.NAME_OR_EMAIL_ERROR);
+    }
+
+    const pepper = parseInt(Math.random() * SERCURITY.PEPPER_RANGE);
+    const userSalt = await bcrypt.hash(getRandString(), SERCURITY.SALT_ROUNDS);
+    const hashedPassword = await bcrypt.hash(
+      composite(password, userSalt, pepper),
+      SERCURITY.SALT_ROUNDS
+    );
+    if (hashedPassword === null) {
+      return respondWithError(res, ERRORS.INTERNAL_ERROR);
+    }
+
+    const newUser = await userService.createOne({
+      name,
+      email,
+      password: hashedPassword,
+      salt: userSalt,
+    });
+    if (!newUser) {
+      return respondWithError(res, ERRORS.INTERNAL_ERROR);
+    }
+    return respondWithStatus(res, CRUD_OPS.CREATED, getPublicUserData(newUser));
+  } catch (err) {
+    return respondWithError(res, ERRORS.INTERNAL_ERROR);
+  }
+};
+
```

Comment:
Instead of repeat yourself you can throw an error and if that's the case go to the catch state: 

For example: 
```js
 try {
    const pepper = Math.floor(Math.random() * SERCURITY.PEPPER_RANGE);
    const userSalt = await bcrypt.hash(getRandString(), SERCURITY.SALT_ROUNDS);
    const hashedPassword = await bcrypt.hash(composite(password, userSalt, pepper), SERCURITY.SALT_ROUNDS);

    if (!hashedPassword)
      throw new Error(ERRORS.INTERNAL_ERROR);

    const newUser = await userService.createOne({ name, email, password: hashedPassword, salt: userSalt });

    if (!newUser)
      throw new Error(ERRORS.INTERNAL_ERROR);

    return respondWithStatus(res, CRUD_OPS.CREATED, getPublicUserData(newUser));
  } catch (err) {
    return respondWithError(res, ERRORS.INTERNAL_ERROR);
  }
};
```